### PR TITLE
Multi–device tests refactoring.

### PR DESCRIPTION
**Pull request by dalexandrov**
Unsupported configuration graceful handling.

Please have a look at this my proposal of a bit more graceful way of handling of unsupported hardware configurations in tests.

Thank you.

### CODE REVIEW

File: unittests/src/main/java/uk/ac/manchester/tornado/unittests/tools/TornadoHelper.java [hunk 1]
```diff
@@ -153,6 +154,14 @@ static void runTestVerbose(String klassName, String methodName) throws ClassNotF
                 bufferFile.append(message);
                 successCounter++;
             } else {
+                // If UnsupportedConfigurationException is thrown this means that test did not fail, it simply can't be run on current configuration
```
Comment (jjfumero) [inline unittests/src/main/java/uk/ac/manchester/tornado/unittests/tools/TornadoHelper.java:157]: Can you pass the auto-formatter? 

https://github.com/beehive-lab/TornadoVM/blob/master/assembly/src/docs/3_INTELLIJ.md

File: unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java [hunk 2]
```diff
@@ -18,24 +18,21 @@
...
 import uk.ac.manchester.tornado.api.TornadoRuntimeCI;
 import uk.ac.manchester.tornado.api.annotations.Parallel;
 import uk.ac.manchester.tornado.api.runtime.TornadoRuntime;
-import uk.ac.manchester.tornado.unittests.common.TornadoTestBase;
+import uk.ac.manchester.tornado.unittests.tools.Exceptions.UnsupportedConfigurationException;
+
+import static org.junit.Assert.*;
 
-public class TestsVirtualLayer extends TornadoTestBase {
+public class TestsVirtualLayer {
```
Comment (jjfumero) [inline unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java:35]: Let's keep this class as an extension of `TornadoTestBase`. We have a clean-up between tests.

File: unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java [hunk 3]
```diff
@@ -65,6 +62,17 @@ public static TornadoRuntimeCI getTornadoRuntime() {
         return TornadoRuntime.getTornadoRuntime();
     }
 
+    /**
+     * Check if enough devices are available
+     */
+    @Before
+    public void enoughDevices(){
```
Comment (jjfumero) [inline unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java:69]: Here you can call `super.before()`?

File: unittests/src/main/java/uk/ac/manchester/tornado/unittests/tools/TornadoHelper.java (hunk 1 above)
Comment (dalexandrov) [reply unittests/src/main/java/uk/ac/manchester/tornado/unittests/tools/TornadoHelper.java:157]: Done!

File: unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java (hunk 2 above)
Comment (dalexandrov) [reply unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java:35]: Done!

File: unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java (hunk 3 above)
Comment (dalexandrov) [reply unittests/src/main/java/uk/ac/manchester/tornado/unittests/virtualization/TestsVirtualLayer.java:69]: Done!

**Pull request merged**